src/engine/threaded_engine.h

```diff
@@ -293,6 +299,24 @@ class ThreadedEngine : public Engine {
     finished_cv_.notify_all();
   }
 
+  /*! \brief Return default OMP thread count. Currently, this is whatever OMP shows as number
+   * of procs
+   */
+  static int DefaultOMPThreadsPerWorker() {
+#ifdef _OPENMP
+    // If environment variable is set and it's not empty, then use omp_get_max_threads()
+    // (Check environment directly, since OMP_NUM_THREADS mnay have odd formatting (i.e. 3, 2"))
+    // Taking max including omp_get_max_threads() in case this implementation of OMP accounts for
+    // hyperthreading
+    const char *s = getenv("OMP_NUM_THREADS");
```

Comment:
We have `dmlc::GetEnv("OMP_NUM_THREADS", 4)` for this, which is cross-platform

Reply:
OMP_NUM_THREADS may be something like "4, 3" to represent top-level and nested OMP thread values.  atoi("4, 3") will return "4", as will dmlc::GetEnv("OMP_NUM_THREADS"), thus nuking (and also incorrectly interpreting) the intended behavior of the set OMP_NUM_THREADS variable.  This is why I explicitly test if the environment variable exists rather than accepting a default, since at that point, I won't know if I am going to override the intended "4, 3" behavior that the user has set.  In  tests since then, I have found that managing both the outer and nested thread counts can affect performance in mxnet.

Reply:
I can use GetEnv() with something nonsensical to test if you prefer, like INT_MIN or something